shared/chat/conversation/list/container.js

```diff
@@ -0,0 +1,157 @@
+  _onLoadMoreMessages: (conversationIDKey: Constants.ConversationIDKey) => { dispatch(Creators.loadMoreMessages(conversationIDKey, false)) },
+  _onRetryMessage: (conversationIDKey: Constants.ConversationIDKey, outboxID: Constants.OutboxIDKey) => dispatch(Creators.retryMessage(conversationIDKey, outboxID)),
+  onDeleteMessage: (message: Constants.Message) => { dispatch(Creators.deleteMessage(message)) },
+  onEditMessage: (message: Constants.Message, body: string) => { dispatch(Creators.editMessage(message, new HiddenString(body))) },
+  onMessageAction: (message: Constants.ServerMessage) => dispatch(navigateAppend([{props: {message}, selected: 'messageAction'}])),
+  onOpenConversation: (conversationIDKey: Constants.ConversationIDKey) => dispatch(Creators.openConversation(conversationIDKey)),
+  onOpenInFileUI: (path: string) => dispatch(({payload: {path}, type: 'fs:openInFileUI'}: OpenInFileUI)),
+  onOpenInPopup: (message: Constants.AttachmentMessage) => dispatch(Creators.openAttachmentPopup(message)),
+  onRetryAttachment: (message: Constants.AttachmentMessage) => dispatch(Creators.retryAttachment(message)),
+})
+
+const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps): Props => {
+  const props = {
+    ...stateProps,
+    ...dispatchProps,
+    headerMessages: List([
+      {key: `chatSecuredHeader-${stateProps.moreToLoad.toString()}`, type: 'ChatSecuredHeader'},
+      {key: `loadingMore-${stateProps.moreToLoad.toString()}`, type: 'LoadingMore'},
+    ]),
+    onLoadAttachment: (messageID, filename) => { stateProps.selectedConversation && dispatchProps._onLoadAttachment(stateProps.selectedConversation, messageID, filename) },
+    onLoadMoreMessages: () => { stateProps.selectedConversation && dispatchProps._onLoadMoreMessages(stateProps.selectedConversation) },
+    onRetryMessage: (outboxID: Constants.OutboxIDKey) => { stateProps.selectedConversation && dispatchProps._onRetryMessage(stateProps.selectedConversation, outboxID) },
+  }
+
+  return {
+    ...props,
+    messages: decorateSupersedes(props.supersedes, props.moreToLoad, props.messages),
+    optionsFn: propsToMessageOptionsFn(props),
+  }
+}
+
+function decorateSupersedes (supersedes, moreToLoad, messages): List<Constants.Message> {
+  if (supersedes && !moreToLoad) {
+    const {conversationIDKey, finalizeInfo: {resetUser}} = supersedes
+    const supersedesMessage: Constants.SupersedesMessage = {
+      key: `supersedes-${conversationIDKey}-${resetUser}`,
+      supersedes: conversationIDKey,
+      timestamp: Date.now(),
+      type: 'Supersedes',
+      username: resetUser,
+    }
+    return messages.unshift(supersedesMessage)
+  }
+
+  return messages
+}
+
+// TODO remove this. Not needed w/ connected messages
+// $FlowIssue wants props with optionsFn inside of it, which this thing itself is building. This'll all go away soon
```

Comment:
annoyingly even though it never references props.optionsFn, flow complains that its missing (which it is, cause this is the thing that's adding it). this'll all go away soon